Sources/DVR/SessionDataTask.swift

```diff
@@ -111,12 +119,17 @@ final class SessionDataTask: URLSessionDataTask {
                 }
             }
 
+            if filteredRequest == nil {
+                persistInteraction = false
+            }
```

Comment:
```suggestion
            persistInteraction = filteredResponse != nil && filteredRequest != nil
```

I would also remove lines 116-117.